src/sentry/tagstore/legacy/backend.py

```diff
@@ -693,7 +696,58 @@ def get_tag_value_qs(self, project_id, environment_id, key, query=None):
         if query:
             queryset = queryset.filter(value__contains=query)
 
-        return queryset
+        return DateTimePaginator(
+            queryset=queryset,
+            order_by=order_by,
+            on_results=lambda results: map(
+                lambda instance: TagValue(
```

Comment:
I there should be a transformation function in here already (see the `transformers` mapping) so you don't have to re-implement this function.